hack/update-codegen.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+../vendor/k8s.io/code-generator/generate-groups.sh all \
```

Comment:
I don't think we should be calling a script in the vendor directory directly here, instead we can use `go run` like so to generate the clients, listeners, informers etc 

Maybe use https://github.com/astoycos/bpfd/blob/client-updates/bpfd-operator/Makefile#L170 as some inspiration? (I did this with most of my inspiration from the Gateway API bits)

Reply:
Done! Also, `make verify` now verifies code generation for clientset related code as well

Reply:
ack, so make verify calls verify-all which calls all the verify-* scripts.. that's useful to keep these things in check even when versions of controller-gen change!